services/src/main/java/org/keycloak/services/clientpolicy/DefaultClientPolicyManager.java

```diff
@@ -158,4 +176,262 @@ private void execute(
+        String validatedJsonString = null;
+        try {
+            validatedJsonString = getValidatedClientPoliciesJson(realm, json);
+        } catch (ClientPolicyException e) {
+            logger.warnv("{0} :: VALIDATE SERIALIZE POLICIES FAILED :: error = {1}, error detail = {2}", LOGMSG_PREFIX, e.getError(), e.getErrorDetail());
+            throw e;
+        }
+        setClientPoliciesJsonString(realm, validatedJsonString);
+        ClientPolicyLogger.logv(logger, "{0} :: UPDATE POLICIES :: realm = {1}, validated and modified PUT = {2}", logMsgPrefix(), realm.getName(), validatedJsonString);
+    }
+
+    @Override
+    public void setupClientPoliciesOnExportingRealm(RealmModel realm, RealmRepresentation rep) {
+        // client profiles  that filter out builtin profiles..
+        ClientProfilesRepresentation filteredOutProfiles = null;
+        try {
+            filteredOutProfiles = getClientProfilesForExport(realm);
+        } catch (ClientPolicyException e) {
+            // set as null
+        }
+        rep.setClientProfiles(filteredOutProfiles);
+
+        // client policies that filter out builtin and policies.
+        ClientPoliciesRepresentation filteredOutPolicies = null;
+        try {
+            filteredOutPolicies = getClientPoliciesForExport(realm);
+        } catch (ClientPolicyException e) {
+            // set as null
+        }
+        rep.setClientPolicies(filteredOutPolicies);
+    }
+
+    @Override
+    public String getClientPolicies(RealmModel realm) {
+        String json = getClientPoliciesJsonString(realm);
+        ClientPolicyLogger.logv(logger, "{0} :: GET POLICIES :: realm = {1}, GET = {2}", logMsgPrefix(), realm.getName(), json);
+        return json;
+    }
+
+    @Override
+    public String getClientProfilesOnKeycloakApp() {
+        return builtinClientProfilesJson;
+    }
+
+    @Override
+    public String getClientPoliciesOnKeycloakApp() {
+        return builtinClientPoliciesJson;
+    }
+
+    @Override
+    public String getClientProfilesJsonString(RealmModel realm) {
+        return realm.getAttribute(CLIENT_PROFILES);
+    }
+
+    @Override
+    public String getClientPoliciesJsonString(RealmModel realm) {
+        return realm.getAttribute(CLIENT_POLICIES);
+    }
+
+    private void setClientProfilesJsonString(RealmModel realm, String json) {
+        realm.setAttribute(CLIENT_PROFILES, json);
+    }
+
+    private void setClientPoliciesJsonString(RealmModel realm, String json) {
+        realm.setAttribute(CLIENT_POLICIES, json);
+    }
+
+    private String getValidatedClientProfilesJson(RealmModel realm, String profilesJson) throws ClientPolicyException {
+        ClientProfilesRepresentation validatedProfilesRep = null;
+        try {
+            validatedProfilesRep = ClientPoliciesUtil.getValidatedClientProfilesRepresentation(session, realm, profilesJson);
+        } catch (ClientPolicyException e) {
```

Comment:
I am not sure what is the point in catch ClientPolicyException and then just throw again ClientPolicyException with exactly same error and errorDetails? Since the method getValidatedClientProfilesJson is declared to throw ClientPolicyException, maybe this try/catch block can be removed entirely? This applies for lots of other places inside this class.

Reply:
@mposolda I think what you pointed out is doing the same thing twice semantically, namely constructing `ClientPolicyException` in `ClientPoliciesUtil.getValidatedClientProfilesRepresentation` and re-constructing this exception as the same contents by the caller of this method.

I agree with your suggestion. I will remove this try/catch block.

Reply:
@mposolda Also, I will find such the other try/catch points and remove them.

Reply:
Thanks!